x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/snapshotlifecycle/SnapshotRetentionConfiguration.java

```diff
@@ -0,0 +1,95 @@
+
+import org.elasticsearch.common.Nullable;
+import org.elasticsearch.common.ParseField;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+import org.elasticsearch.common.io.stream.Writeable;
+import org.elasticsearch.common.unit.TimeValue;
+import org.elasticsearch.common.xcontent.ConstructingObjectParser;
+import org.elasticsearch.common.xcontent.ToXContentObject;
+import org.elasticsearch.common.xcontent.XContentBuilder;
+import org.elasticsearch.common.xcontent.XContentParser;
+
+import java.io.IOException;
+import java.util.Objects;
+
+public class SnapshotRetentionConfiguration implements ToXContentObject, Writeable {
+
+    public static final SnapshotRetentionConfiguration EMPTY = new SnapshotRetentionConfiguration((TimeValue) null);
+
+    private static final ParseField EXPIRE_AFTER = new ParseField("expire_after");
+
+    private static final ConstructingObjectParser<SnapshotRetentionConfiguration, Void> PARSER =
+        new ConstructingObjectParser<>("snapshot_retention", true, a -> {
+            TimeValue expireAfter = a[0] == null ? null : TimeValue.parseTimeValue((String) a[0], EXPIRE_AFTER.getPreferredName());
+            return new SnapshotRetentionConfiguration(expireAfter);
+        });
+
+    static {
+        PARSER.declareString(ConstructingObjectParser.optionalConstructorArg(), EXPIRE_AFTER);
+    }
+
+    // TODO: add the rest of the configuration values
+    private final TimeValue expireAfter;
+
+    public SnapshotRetentionConfiguration(@Nullable TimeValue expireAfter) {
+        this.expireAfter = expireAfter;
+    }
+
+    public SnapshotRetentionConfiguration(StreamInput in) throws IOException {
+        this.expireAfter = in.readOptionalTimeValue();
+    }
+
+    public static SnapshotRetentionConfiguration parse(XContentParser parser, String name) {
+        return PARSER.apply(parser, null);
+    }
+
+    public TimeValue getExpireAfter() {
+        return this.expireAfter;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (expireAfter != null) {
+            builder.field(EXPIRE_AFTER.getPreferredName(), expireAfter.getStringRep());
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalTimeValue(this.expireAfter);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(expireAfter);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+        if (obj == null) {
+            return false;
+        }
+        if (obj.getClass() != getClass()) {
+            return false;
+        }
+        SnapshotRetentionConfiguration other = (SnapshotRetentionConfiguration) obj;
+        return this.expireAfter.equals(other.expireAfter);
```

Comment:
`this.expireAfter` can be null, so this should use `Objects.equals()` to avoid throwing an NPE.

Reply:
Good catch, thanks